packages/presentation/lib/screens/login/login_bloc.dart

```diff
@@ -43,23 +56,44 @@ class _LoginBloc extends BlocImpl<BaseArguments, LoginData>
   final FacebookAuthUseCase _facebookAuthUseCase;
   final GoogleAuthUseCase _googleAuthUseCase;
   final SaveCredentialsUseCase _saveCredentialsUseCase;
+  final LoginValidationUseCase _loginValidationUseCase;
+  final LoginViewMapper _loginViewMapper;
 
   final TextEditingController _loginController = TextEditingController();
   final TextEditingController _passwordController = TextEditingController();
 
+  final _formStateGlobalKey = GlobalKey<FormState>();
+
   _LoginBloc(
     this._userIsRegisteredUseCase,
     this._facebookAuthUseCase,
     this._googleAuthUseCase,
     this._saveCredentialsUseCase,
+    this._loginValidationUseCase,
+    this._loginViewMapper,
   ) : super(initState: const LoginData.init());
 
+  @override
+  void initState() {
+    super.initState();
+
+    _loginController.addListener(_resetErrorMessages);
+
+    _passwordController.addListener(_resetErrorMessages);
```

Comment:
don't reset messages for both fields at the same time. reset validation only for the field which was changed